stl/inc/vector

```diff
@@ -668,15 +736,16 @@ private:
     }
 
 public:
-    vector& operator=(vector&& _Right) noexcept(noexcept(_Move_assign(_Right, _Choose_pocma<_Alty>{}))) {
+    _CONSTEXPR20_DYNALLOC vector& operator=(vector&& _Right) noexcept(
+        noexcept(_Move_assign(_Right, _Choose_pocma<_Alty>{}))) {
         if (this != _STD addressof(_Right)) {
             _Move_assign(_Right, _Choose_pocma<_Alty>{});
         }
 
         return *this;
     }
 
-    ~vector() noexcept {
+    _CONSTEXPR20_DYNALLOC ~vector() noexcept {
         _Tidy();
 #if _ITERATOR_DEBUG_LEVEL != 0

```

Comment:
Yeah, I did no understand the inner workings of the iterator proxies good enough when writing this.

While this has not really changed, I believe with the new test coverage I would be more confident to make those changes

Reply:
So looking at this a bit more I found a lot of cases like:
```cpp
        auto&& _Alproxy = _GET_PROXY_ALLOCATOR(_Alty, _Getal());
        _Container_proxy_ptr<_Alty> _Proxy(_Alproxy, _Mypair._Myval2);
        
        // Do something

        _Proxy._Release();
```

It would be quite an invasive change to factor all places into sub functions, so I am not sure whether this is something I would like to do in the bringup or postpone for a later PR, especialy given that there are plans to rework proxies for vNext anyway.

Reply:
Removed all allocations during constant evaluations